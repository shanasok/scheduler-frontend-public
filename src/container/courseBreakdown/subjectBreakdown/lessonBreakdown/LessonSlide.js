import React from 'react';
// import * as PropTypes from "prop-types";
import { connect } from "react-redux";
import {withRouter, Redirect} from "react-router-dom";
import SlideHeader from "./slide/SlideHeader"
import SlideNumbers from "./slide/SlideNumbers";
import SlideChooser from "./slide/SlideChooser";

import '../../../../css/lessonStyles.css';
import {bindActionCreators, compose} from "redux";
import {setCurrentPage} from "../../../../redux/actions/pageAction";
import {setCurrentLessonContent} from "../../../../redux/actions/lessonContentAction";
import {getLessonContent} from "../../../../utils/lessonContentFetcherGraphQL";

class LessonSlide extends React.Component {

    componentDidMount() {
        getLessonContent(this.props.lesson.id, this.props.actions.setCurrentLessonContent);
    }

    render() {
        const lesson = this.props.lesson;
        if (lesson.pages === undefined) {
            return <Redirect to="/"/>
        }else {
            const currentPage = this.props.currentPage;
            const pageCountTotal = lesson.pages.length;
            const currentPageNumber = currentPage.pageNumber;

            return <div className="lessonSlide">
                <SlideHeader slideHeader={lesson.name}/>
                <SlideNumbers currentPageNumber={currentPageNumber} totalSlidesForLesson={pageCountTotal}/>
                <SlideChooser currentPage={currentPage} />
            </div>

        }
    }
}

const mapStateToProps = state => ({
    lesson: state.currentLesson,
    currentPage: state.currentPage,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setCurrentPage,
            setCurrentLessonContent,
        }, dispatch)
    }
}

export default compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps)
)(LessonSlide);

// LessonSlide.propTypes = {
//     location: PropTypes.shape({
//         state: PropTypes.shape({
//             lesson: PropTypes.shape({
//                 id: PropTypes.number.isRequired,
//                 name: PropTypes.string,
//                 lessonNumber: PropTypes.number,
//                 pages: PropTypes.array,
//             }).isRequired
//         })
//     })
// };