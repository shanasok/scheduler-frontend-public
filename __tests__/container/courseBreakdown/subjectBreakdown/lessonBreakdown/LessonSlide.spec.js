import React from 'react';
import LessonSlide from '../../../../../src/container/courseBreakdown/subjectBreakdown/LessonBreakdown/LessonSlide';
import renderer from 'react-test-renderer';
import {Router} from "react-router-dom";
import { createMemoryHistory } from "history";
import {Provider as ReduxProvider} from "react-redux";
import configureMockStore from "redux-mock-store";

const currentPage = {
    id: 1,
    pageNumber: 1,
};

const lesson = {
    id: 1,
    name: "test lesson name",
    lessonNumber: 1,
    pages: [currentPage],
};

const mockStore = configureMockStore()
const mockState = {
    currentLesson: lesson,
    currentPage: currentPage,
}
let store = mockStore(mockState)

it('renders correctly', () => {

    const history = createMemoryHistory();
    history.location.state = {
        lesson: {
            id: 1,
            name: "Threading the Machine",
            lessonNumber: 1,
            pages:[],
        }
    };

    const tree = renderer
        .create(<ReduxProvider store={store}><Router history={history}><LessonSlide/></Router></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('redirects when lesson has no pages', () => {

    const history = createMemoryHistory();
    history.location.state = {
        lesson: {
            id: 1,
            name: "Threading the Machine",
            lessonNumber: 1,
        }
    };

    const lesson = {
        id: 1,
        name: "test lesson name",
        lessonNumber: 1,
    };

    const mockStore = configureMockStore()
    const mockState = {
        currentLesson: lesson,
        currentPage: currentPage,
    }
    let store = mockStore(mockState)

    const tree = renderer
        .create(<ReduxProvider store={store}><Router history={history}><LessonSlide/></Router></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
