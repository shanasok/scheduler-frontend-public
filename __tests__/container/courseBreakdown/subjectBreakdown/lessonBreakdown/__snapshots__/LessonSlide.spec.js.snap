// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects when lesson has no pages 1`] = `null`;

exports[`renders correctly 1`] = `
<div
  className="lessonSlide"
>
  <div
    className="lessonTitle"
  >
    test lesson name
    <div>
      <button
        className="curvedButton"
        onClick={[Function]}
      >
        Edit Slide
      </button>
    </div>
  </div>
  <div
    className="pageNumbers"
  >
    Page 
    1
     of 
    1
  </div>
  <div
    className="slideChooser"
  >
    <div
      className="arrow inactiveLeftArrow"
    />
    <div
      className="slideBox"
    >
      No content on this slide yet
    </div>
    <div
      className="arrow inactiveRightArrow"
    />
  </div>
</div>
`;
